google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1563,7 +1605,7 @@ def post_object path, policy: nil, issuer: nil,
                         private_key: nil
           ensure_service!
 
-          signer = File::Signer.from_bucket self, path
+          signer = File::SignerV2.from_bucket self, path
```

Comment:
Does `PostObject` have any behavior difference for `v2` vs. `v4`?

Reply:
@frankyn and I reviewed and discussed `PostObject` this week. My understanding from him is that it is a separate use case that does not fall under V4. (Probably it does not belong in a class named `SignerV2` either.) @frankyn: any thoughts on this?

Reply:
AFAIK the `PostObject` constructor being part of `Signer` (now `SignerV2`) is an internal implementation detail, and not part of the public API. It can be refactored without breaking the public API. This can be done in this PR, a separate PR, or not at all.

Reply:
V4 will have a post form policy but it's not supported right now by the backend.

Reply:
Thanks @frankyn. It sounds like we can leave things as they are for this question.